cycler.py

```diff
@@ -371,6 +375,9 @@ def _transpose(self):
                 out[k].append(d[k])
         return out
 
+    # for back compatibility
+    _transpose = by_key
```

Comment:
I was 50/50 on this.  I was thinking of being semi-evil and using the private `_transpose` in mpl so that it will work with current tagged version of cycler so we don't have to do a full release cycle on `cycler` before implementing the key introspection on mpl.